boot/bootutil/src/loader.c

```diff
@@ -1239,36 +1239,70 @@ boot_validated_swap_type(struct boot_loader_state *state,
  * @param off                   The offset within the flash area to start the
  *                              erase.
  * @param size                  The number of bytes to erase.
+ * @param backwards             If set to true will erase from end to start
+ *                              addresses, otherwise erases from start to end
+ *                              addresses.
  *
  * @return                      0 on success; nonzero on failure.
  */
 int
-boot_scramble_region(const struct flash_area *fa, uint32_t off, uint32_t size)
+boot_scramble_region(const struct flash_area *fa, uint32_t off, uint32_t size, bool backwards)
 {
-    int ret = 0;
+    int rc = 0;
 
     if (size == 0) {
-        return 0;
+        goto done;
+    } else if (off >= flash_area_get_size(fa) || (flash_area_get_size(fa) - off) < size) {
+        rc = -1;
+        goto done;
```

Comment:
Want to add the same check to `boot_erase_region`?

Reply:
Added. Think mynewt might be broken with this PR...

Reply:
> Added. Think mynewt might be broken with this PR...

What will break it?

Reply:
CI was running for 4 hours on mynewt, I assume it just locked up in some loop, what broke it though I have no clue

Reply:
If that keeps happening I will try to run this locally, should still have mynewet env available.